tao_compiler/mlir/disc/transforms/fusion_utils.cc

```diff
@@ -486,69 +660,197 @@ void FusionPattern::calculateOperandsAndResults() {
   for (Value v : input_set) operands_.push_back(v);
 }
 
-// Supports using EquivalenceClasses for Value
-bool operator<(const ValueWrapper& lhs, const ValueWrapper& rhs) {
-  auto lhs_value = lhs.getValue().getAsOpaquePointer();
-  auto rhs_value = rhs.getValue().getAsOpaquePointer();
-  return lhs_value < rhs_value;
+// Create a new fusion pattern from a single op.
+FusionPattern::FusionPattern(Operation* op) : FusionPatternBase(op) {
+  if (isRank2RowReduction(op)) {
+    fusion_type_ = FusionType::kRowReduction;
+  } else if (isRank2ColReduction(op)) {
+    fusion_type_ = FusionType::kColReduction;
+  } else if (mlir::disc_ral::isFusible(op)) {
```

Comment:
disc-cpu may have different supported op list than disc-gpu, thus directly call isFusible is generally not a good idea. We prefer to init a fusion pattern using the initFusionPattern method of a fusion strategy.

As for this specific constructor, I suggests just to follow original fusionPattern design.